src/i18n/mod.rs

```diff
@@ -0,0 +1,19 @@
+extern crate alloc;
+use alloc::string::{String, ToString};
+use unicode_bidi::BidiInfo;
+
+mod arabic;
+
+/// Preprocess a line of text so that it may be rendered via Unifont.
+/// Currently supported scripts: Arabic.
+pub fn preprocess_line(line: &str) -> String {
+    let text = arabic::use_contextual_forms(line);
+
+    // apply the BiDi algorithm, assuming that the text is a single line
+    let bidi_info = BidiInfo::new(&text, None);
+    let para = &bidi_info.paragraphs[0];
+    let line = para.range.clone();
+    let display = bidi_info.reorder_line(para, line);
```

Comment:
@Konosh93 What do you think about an approach where we only provide the `i18n::preprocess_text` function that does not intend to deal with BiDi, and depend on `unicode-bidi` only in `examples/banner.rs` to show how it's meant to be used?

Reply:
I think it is a good approach. We can add this implicit support later once we figure out how to port the bidi algorithm without the std library. 👍

Reply:
Turns out `unicode-bidi` is already `no_std` + `alloc` compatible, it just needs the proper feature setup! :)

Reply:
In any case, what bothers me about this BiDi stuff is that it isn't clear how the user would control the breaking of the text into multiple lines, without a clear understanding of what's happening.

So for now, let's postpone figuring that out. I'll move this BiDi stuff into the example code, and explain that dealing with the BiDi aspect of the text is the responsibility of the user for now...

Reply:
Sounds good.